### DISCUSSION TITLE
Project 4 Question 1

### DISCUSSION THREAD

**Discussion by FishMacaron** [cooperrc/computational-mechanics#72] [Q&A] [answered]
These are open ended questions, correct? It doesn't sound like these need code to be answered. I'm just confused as to where I should be looking in the modules to get help come to a conclusion. Even when I try searching for key words in the modules I can't seem to find the information I need to understand these questions completely. 
I'm also open to outside sources I can use to look into the topics covered in these questions.

**Specifically, I am having issues with the following for each question:**

a and c) I'm confused as to what error it may be referring to. Is it describing the error, like saying it will be a large or small number? Or is there certain types of errors I should be thinking about like how truncation can lead to differences in the actual answers vs the ones in code? Those are my initial thoughts when I look at these questions but none of them sound quite right.

b) I can't find out the importance of the problem being "underconstrained" or what BC represents (assuming it's an acronym).

> a. What error would you expect when you solve for u in K*u = F?

> b. Why is the condition of K so large? The problem is underconstrained. It describes stiffness of structure, but not the BC's. So, we end up with sumF=0 and -sumF=0

> c. What error would you expect when you solve for u[2:13] in K[2:13,2:13]*u=F[2:13]

**Comment by cooperrc** [comment, marked as answer]
These should have an _order of magnitude_ answer e.g. $10^0$ or  $10^{-12}$

The [condition](https://cooperrc.github.io/computational-mechanics/module_04/01_Linear-Algebra.html#condition-of-a-matrix) gives you the maximum expected error when you do Gauss elimination, LU decompostion, or any other pseudo inverse method.

**Reply by FishMacaron** [reply]
Does np.linalg.cond() give you that condition number without having to use the def for a pseudo inverse method? If not, what is it actually providing because its purpose is to "[[return] the condition number](https://numpy.org/doc/stable/reference/generated/numpy.linalg.cond.html#:~:text=cond,-linalg.&text=Compute%20the%20condition%20number%20of,p%20(see%20Parameters%20below).)".  and if the "the condition gives you the maximum expected error" using that should give me the answer, right? Unless those two aren't the same thing...
